src/components/Editor/DurationSelector.vue

```diff
@@ -0,0 +1,403 @@
+	/** Maximum days for custom duration */
+	maxDays?: number
+	/** Maximum hours for custom duration */
+	maxHours?: number
+	/** Step for minutes selector */
+	minuteStep?: number
+}
+
+const props = withDefaults(defineProps<Props>(), {
+	modelValue: 0,
+	format: 'minutes',
+	label: () => t('calendar', 'Duration'),
+	labelOutside: false,
+	predefinedOptions: () => [
+		{ value: 15, label: t('calendar', '15 minutes') },
+		{ value: 30, label: t('calendar', '30 minutes') },
+		{ value: 60, label: t('calendar', '1 hour') },
+		{ value: 90, label: t('calendar', '1 hour 30 minutes') },
+		{ value: 120, label: t('calendar', '2 hours') },
+	],
+	showCustom: true,
+	showDays: true,
+	showHours: true,
+	showMinutes: true,
+	maxDays: 30,
+	maxHours: 23,
+	minuteStep: 5,
+})
+
+const emit = defineEmits<{
+	'update:model-value': [value: number | string]
+}>()
+const internalSelectedOption = ref<SelectOption | null>(null)
+const customDays = ref<SelectOption | null>(null)
+const customHours = ref<SelectOption | null>(null)
+const customMinutes = ref<SelectOption | null>(null)
+
+// Computed
+
+// Normalize modelValue to minutes
+const internalMinutes = computed<number>(() => {
+	if (typeof props.modelValue === 'string') {
+		return iso8601ToMinutes(props.modelValue)
+	}
+	return props.modelValue || 0
+})
+
+const allOptions = computed<SelectOption[]>(() => {
+	const options = props.predefinedOptions.map((opt: DurationOption) => ({
+		id: opt.value,
+		label: opt.label,
+	}))
+
+	if (props.showCustom) {
+		options.push({
+			id: 'custom',
+			label: t('calendar', 'Custom'),
+		})
+	}
+
+	return options
+})
+
+const selectedOption = computed<SelectOption | null>({
+	get() {
+		// If we have an internal selection, use it
+		if (internalSelectedOption.value) {
+			return internalSelectedOption.value
+		}
+
+		// Otherwise, calculate from modelValue
+		const predefined = props.predefinedOptions.find((opt: DurationOption) => opt.value === internalMinutes.value)
+		if (predefined) {
+			const option = {
+				id: predefined.value,
+				label: predefined.label,
+			}
+			internalSelectedOption.value = option
+			return option
+		}
+
+		// Only select custom if we have a non-zero duration that doesn't match predefined options
+		if (props.showCustom && internalMinutes.value > 0) {
+			const customOption = {
+				id: 'custom',
+				label: t('calendar', 'Custom'),
+			}
+			internalSelectedOption.value = customOption
+			return customOption
+		}
+
+		return null
+	},
+	set(value: SelectOption | null) {
+		internalSelectedOption.value = value
+	},
+})
+
+const isCustomSelected = computed<boolean>(() => {
+	return selectedOption.value?.id === 'custom'
+})
+
+const daysOptions = computed<SelectOption[]>(() => {
+	const options: SelectOption[] = []
+	for (let i = 0; i <= props.maxDays; i++) {
+		options.push({
+			id: i,
+			label: i === 0 ? t('calendar', '0 days') : i === 1 ? t('calendar', '1 day') : t('calendar', '{count} days', { count: i }),
+		})
```

Comment:
Don't assume a plural function. Use the existing l10n mechanism that come with a plural function for every language.